Add missing SC on skills properties

illzbee opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

"Closes #54"
Adds missing SC on skills properties
Used https://github.com/Ivaar/Skillchains/blob/master/skills.lua for reference and ffxiclopedia to double check on removal/addition to "main/sub skillchains"

illzbee committed:
Squashed commit of the following:

commit 7cb170a4ce02c94867ed1ee13a83df5d8031c174
Author: Illzbee <wletour@gmail.com>
Date:   Sat Aug 13 17:45:32 2022 -0400

    Squashed commit of the following:

    commit 1443cd37a3cbfd85b72726bc2d69752d49ef9a48
    Merge: bf5b569eeb 36e95d9cf4
    Author: claywar <81713309+claywar@users.noreply.github.com>
    Date:   Sat Aug 13 15:49:15 2022 -0400

        Merge pull request #2518 from LandSandBoat/interaction/forget-me-not

        Interaction - Missions: Forget Me Not through When Wills Collide

    commit 36e95d9cf4585903d5676e79eca5b7fe66732afc
    Author: claywar <claywar@beerlab.dev>
    Date:   Sat Aug 13 15:28:38 2022 -0400

        Interaction - Mission: When Wills Collide

    commit bfd72b951f99aca85497baeb19818ef74e83743f
    Author: claywar <claywar@beerlab.dev>
    Date:   Sat Aug 13 13:26:53 2022 -0400

        Interaction - Mission: Time Slips Away

    commit b9809997100e441f6490b75c16c8bb678190550e
    Author: claywar <claywar@beerlab.dev>
    Date:   Sat Aug 13 12:09:56 2022 -0400

        Interaction - Mission: Glimmer of Life

    commit 8062c414fa86fabc4bdec267eafd156f97dbd444
    Author: claywar <claywar@beerlab.dev>
    Date:   Sat Aug 13 11:25:10 2022 -0400

        Interaction - Mission: Pillar of Hope

    commit 1d9d21a5b920163386c175bdae1a62eeba62904f
    Author: claywar <claywar@beerlab.dev>
    Date:   Sat Aug 13 09:32:20 2022 -0400

        Interaction - Mission: Forget Me Not

commit 72630ce2dd0b2dcd84547e99bfd5146925bd34a0
Author: Illzbee <wletour@gmail.com>
Date:   Sat Aug 13 17:33:26 2022 -0400

    Update blue_spell_list.sql

commit d271fc497961e56a49c56ac068cc84233338d875
Author: Illzbee <wletour@gmail.com>
Date:   Sat Aug 13 17:02:25 2022 -0400

    Update weapon_skills.sql

commit 79f10980c617763add1a6449929cdda337a7dc59
Author: Illzbee <wletour@gmail.com>
Date:   Sat Aug 13 16:46:48 2022 -0400

    Add missing SC on skills properties

    Add missing SC on skills properties

WinterSolstice8 commented:
Changelist is huge for BLU spells, which ones added a tertiary SC property?

WinterSolstice8 commented:
WS properties check out

zach2good commented:
For the BLU changes, are you going to hook up reading the new SQL column in core?

https://github.com/LandSandBoat/server/blob/d9d3a7b6494d17f2178b140c9b0f6d9e983d7459/src/map/spell.cpp#L541

illzbee commented:
> Changelist is huge for BLU spells, which ones added a tertiary SC property?

Sorry should have mentioned only one added a tertiary sc "Quadrastrike" rest was just removing ' '

illzbee committed:
new SQL column BLU

Set new SQL column reading tertiary skillchain

illzbee commented:
> For the BLU changes, are you going to hook up reading the new SQL column in core?
> 
> https://github.com/LandSandBoat/server/blob/d9d3a7b6494d17f2178b140c9b0f6d9e983d7459/src/map/spell.cpp#L541

Should be good to go now. Thanks Zach!

illzbee committed:
content tag

char* contentTag = (char*)sql->GetData(7); <<<to>>> char* contentTag = (char*)sql->GetData(8);

illzbee committed:
Update spell.cpp

illzbee committed:
Update spell.cpp

illzbee committed:
Update spell.cpp

illzbee committed:
Merge branch 'AddSC' of https://github.com/illzbee/LSB into AddSC

WinterSolstice8 commented:
`no member named 'setTertiarySkillchain' in 'CBlueSpell';`
that means all the plumbing for tertiary SC effect is missing.  you'd have to add all the calls to get/set that during skillchains to make it properly work. If you'd like, you could PR just the SQL changes and we can handle the C++ part. You could use this as a learning experience as well.

illzbee committed:
Update blue_spell.h

add getsetTertiarySkillchain

illzbee committed:
Update spell.cpp

illzbee committed:
Update blue_spell.h

zach2good commented:
You've added your declaration for your new methods in `blue_spell.h`, but you haven't added any definitions in `blue_spell.cpp`

https://stackoverflow.com/questions/1410563/what-is-the-difference-between-a-definition-and-a-declaration

Like Primary has these in the header file:
```cpp
    uint8  getPrimarySkillchain() const;
    void   setPrimarySkillchain(uint8 sc);
```

and these in the cpp file:
```cpp

uint8 CBlueSpell::getPrimarySkillchain() const
{
    return m_PrimarySkillchain;
}

void CBlueSpell::setPrimarySkillchain(uint8 sc)
{
    m_PrimarySkillchain = sc;
}
```

illzbee committed:
Update blue_spell.cpp

illzbee commented:
> You've added your declaration for your new methods in `blue_spell.h`, but you haven't added any definitions in `blue_spell.cpp`
> 
> https://stackoverflow.com/questions/1410563/what-is-the-difference-between-a-definition-and-a-declaration
> 
> Like Primary has these in the header file:
> 
> ```c++
>     uint8  getPrimarySkillchain() const;
>     void   setPrimarySkillchain(uint8 sc);
> ```
> 
> and these in the cpp file:
> 
> ```c++
> uint8 CBlueSpell::getPrimarySkillchain() const
> {
>     return m_PrimarySkillchain;
> }
> 
> void CBlueSpell::setPrimarySkillchain(uint8 sc)
> {
>     m_PrimarySkillchain = sc;
> }
> ```

yeah I just did this ran a search on all src files for primary to find where I was missing ...lol... thanks again.

zach2good commented on src/map/blue_spell.cpp:
```diff
@@ -103,6 +103,16 @@ void CBlueSpell::setSecondarySkillchain(uint8 sc)
     m_SecondarySkillchain = sc;
 }
 
+uint8 CBlueSpell::getSecondarySkillchain() const
+{
+    return m_TertiarySkillchain;
+}
+
+void CBlueSpell::setSecondarySkillchain(uint8 sc)
```
Tertiary *

illzbee replied:
Did I spell something incorrect?

illzbee replied:
oh duh lol dont mind me

illzbee replied:
rushing before work isn't always good

illzbee committed:
Update blue_spell.cpp
